src/Doctrine/Orm/Filter/FilterExpressionProvider.php

```diff
@@ -16,23 +17,35 @@ class FilterExpressionProvider implements DoctrineOrmExpressionProviderInterface
      */
     protected array $filters;
 
+    private bool $innerJoinsLeft;
+
     /**
      * @param ContextAwareFilterInterface[] $filters
      */
-    public function __construct(array $filters)
+    public function __construct(array $filters, bool $innerJoinsLeft = false)
     {
         $this->filters = $filters;
+        $this->innerJoinsLeft = $innerJoinsLeft;
     }
 
     public function getFunctions(): array
     {
-        return array_map(static function ($filter) {
+        $innerJoinsLeft = $this->innerJoinsLeft;
+        return array_map(static function ($filter) use ($innerJoinsLeft) {
             return new ExpressionFunction(
                 self::normalizeFilterName(get_class($filter)),
                 static function () {
                     // Uncompilable
                 },
-                static function ($arguments, string $property = null, string $strategy = null, $value = null) use ($filter) {
+                static function (
+                    $arguments,
+                    string $property = null,
+                    string $strategy = null,
+                    $value = null
+                ) use (
+                    $filter,
+                    $innerJoinsLeft
+                ) {
```

Comment:
Add a $innerJoinsLeft to the parameters with a default value of "false" and remove the class property to allow a user to change the behaviour of the innerJoins only in that filter.